packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts

```diff
@@ -517,6 +522,47 @@ export class StripeHandler {
     return couponDetails;
   }
 
+  /**
+   * Updates customer subscription with promotion code
+   */
+  async applyPromotionCodeToSubscription(request: AuthRequest) {
+    this.log.begin('subscriptions.applyPromotionCodeToSubscription', request);
+
+    const { uid, email } = await handleAuth(this.db, request.auth, true);
+    await this.customs.check(
+      request,
+      email,
+      'applyPromotionCodeToSubscription'
+    );
+
+    const customer = await this.stripeHelper.fetchCustomer(uid);
+    if (!customer) {
+      throw error.unknownCustomer(uid);
+    }
+
+    const { promotionId, subscriptionId } = request.payload as Record<
+      string,
+      string
+    >;
+
+    try {
+      const updatedSubscription =
+        await this.stripeService.applyPromoCodeToSubscription(
+          customer.id,
+          subscriptionId,
+          promotionId
+        );
+
+      return updatedSubscription;
+    } catch (error) {
+      this.log.error('subscriptions.applyPromotionCodeToSubscription', {
+        error,
+        uid,
+      });
+      throw error;
```

Comment:
This needs to throw an error from the `errors` library that is transformable into the appropriate HTTP response code. Throwing the library here will result in a 500 error from hapi and this will end up in Sentry.